Kuadrant/authorino-operator
# Adds preflight checks before install the operator

This PR adds preflight checks to the operator to validate required resources are created prior the installation of an Authorino instances.

The first check added verifies whether TLS is enabled and the secrets with the certs are created

## Verification steps
1. Install and deploy the operator from this branch
2. Create an instance of authorino with TLS enabled
```bash
kubectl create namespace myapp 

kubectl -n myapp apply -f -<<EOF
apiVersion: operator.authorino.kuadrant.io/v1beta1
kind: Authorino
metadata:
  name: authorino
spec:
  replicas: 1
  clusterWide: false
  logLevel: info
  logMode: production
  listener:
    tls:
      enabled: true
      certSecretRef:
        name: ""
  oidcServer:
    tls:
      enabled: true
      certSecretRef:
        name: ""
EOF
```
3. Verify if the installation returns an error
```bash
kubectl get authorino authorino -n myapp -o yaml
status:
  conditions:
  - lastTransitionTime: "2022-01-11T16:03:15Z"
    message: 'oidc secret name test not found, err: Secret "test" not found'
    reason: TlsSecretNotProvided
    status: "False"
    type: Ready
```
